src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -193,42 +189,14 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
   }
 
   async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
-    let business = null
+    let tingBusiness = await this.fetchTingInfo(businessLookup)
 
-    // Get the amalgamating business information, mailing address, and email if in LEAR.
-    // Otherwise, return error.
-    const data = await Promise.all([
-      LegalServices.fetchBusinessInfo(businessLookup.identifier),
-      AuthServices.fetchAuthInfo(businessLookup.identifier),
-      LegalServices.fetchAddresses(businessLookup.identifier)
-    ]).catch((error) => {
-      return error
-    })
-
-    if (data.length === 3) {
-      business = data[0].data?.business
-      business.businessContact = data[1].contacts[0]
-      business.officeAddress = data[2]
-    }
-
-    // If the business is not null (LEAR Entity), create from it a TING business following the interface.
     // If the amalgamating businesses array is not empty, check if identifier already exists.
     // If identifier already exists, don't add the business to the array.
-    if (business) {
+    if (tingBusiness) {
       const amalgamatingBusinesses = this.getAmalgamatingBusinesses
 
-      const tingBusiness = {
-        type: 'lear',
-        role: AmlRoles.AMALGAMATING,
-        identifier: business.identifier,
-        name: business.legalName,
-        email: business.businessContact.email,
-        legalType: business.legalType,
-        address: business.officeAddress.registeredOffice.mailingAddress,
-        goodStanding: business.goodStanding
-      } as AmalgamatingBusinessIF
-
-      if (!amalgamatingBusinesses.find((b: any) => b.identifier === business.identifier)) {
+      if (!amalgamatingBusinesses.find((b: any) => b.identifier === (tingBusiness as any).identifier)) {
```

Comment:
Had to add the any here because VS code wasn't happy:
![Screenshot (121)](https://github.com/bcgov/business-create-ui/assets/122301442/cd15886d-ea14-482c-bba7-4cef8f2496fb)

Reply:
This is that type error due to VS Code not knowing which of the 2 union types you're using. `any` is OK for now.